src/modules/products/products.service.ts

```diff
@@ -200,4 +201,53 @@ export class ProductsService {
       data: {},
     };
   }
+
+  private getDateRange(date: Date) {
+    const monthStarts = startOfMonth(new Date(date));
+    const monthEnds = endOfMonth(new Date(date));
+
+    return { monthStarts, monthEnds };
+  }
+
+  private async getTotalProductsForDateRange(startOfMonth: Date, endOfMonth: Date) {
+    const result = await this.productRepository
+      .createQueryBuilder('product')
+      .select('SUM(product.quantity)', 'total')
+      .where('product.created_at BETWEEN :startOfMonth AND :endOfMonth', { startOfMonth, endOfMonth })
+      .getRawOne();
+
+    return result && result.total ? parseInt(result.total, 10) : 0;
+  }
+
+  async getTotalProducts() {
+    const todaysDate = new Date();
+    const lastMonth = subMonths(todaysDate, 1);
+
+    const monthStarts = this.getDateRange(todaysDate).monthStarts;
+    const monthEnds = this.getDateRange(todaysDate).monthEnds;
+    const lastMonthStarts = this.getDateRange(lastMonth).monthStarts;
+    const lastMonthEnds = this.getDateRange(lastMonth).monthEnds;
+
+    const totalProductsThisMonth = await this.getTotalProductsForDateRange(monthStarts, monthEnds);
+
+    const totalProductsLastMonth = await this.getTotalProductsForDateRange(lastMonthStarts, lastMonthEnds);
+
+    let percentageChange = '0% from last month';
+    if (totalProductsLastMonth !== 0) {
+      const change = ((totalProductsThisMonth - totalProductsLastMonth) / totalProductsLastMonth) * 100;
+      percentageChange = `${change > 0 ? '+' : ''}${change.toFixed(2)}% from last month`;
+    } else if (totalProductsThisMonth > 0) {
+      percentageChange = `+100.00% from last month`;
+    }
```

Comment:
Add an else statement for when there is no product last month and this month

```suggestion
percentageChange = `No change from last month`;
```